lib/app/features/core/permissions/strategies/base_permission_strategy.dart

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:ice/app/features/core/permissions/data/models/permissions_types.dart';
+import 'package:ice/app/features/core/permissions/strategies/permission_strategy.dart';
+import 'package:permission_handler/permission_handler.dart' as ph;
+
+abstract class BasePermissionStrategy implements PermissionStrategy {
```

Comment:
Do we need `PermissionStrategy` interface? We can replace the interface with this abstract class.

Reply:
`PermissionStrategy` is needed for the platform-specific cases where `permission_handler` isn’t applicable